Tests/tests/test_FrontPanel.cpp

```diff
@@ -0,0 +1,523 @@
+    ON_CALL(frontPanelConfigImplMock, getIndicators())
+        .WillByDefault(::testing::Return(device::List<device::FrontPanelIndicator>({indicatorList})));
+    ON_CALL(frontPanelIndicatorMock, getName())
+        .WillByDefault(::testing::Return("red"));
+
+    ASSERT_TRUE(dsFrontPanelModeChange != nullptr);
+
+
+    IARM_Bus_PWRMgr_EventData_t eventData;
+    eventData.data.state.newState =IARM_BUS_PWRMGR_POWERSTATE_ON;
+    eventData.data.state.curState =IARM_BUS_PWRMGR_POWERSTATE_STANDBY;
+
+    handler.Subscribe(0, _T("powerModeChange"), _T("client.events.powerModeChange"), message);
+
+    dsFrontPanelModeChange(IARM_BUS_PWRMGR_NAME, IARM_BUS_PWRMGR_EVENT_MODECHANGED, &eventData , 0);
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("powerLedOn"), _T("{\"index\": \"power_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("powerLedOn"), _T("{\"index\": \"record_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("powerLedOn"), _T("{\"index\": \"data_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+
+
+    eventData.data.state.newState =IARM_BUS_PWRMGR_POWERSTATE_OFF;
+    eventData.data.state.curState =IARM_BUS_PWRMGR_POWERSTATE_STANDBY;
+
+    handler.Subscribe(0, _T("powerModeChange"), _T("client.events.powerModeChange"), message);
+
+    dsFrontPanelModeChange(IARM_BUS_PWRMGR_NAME, IARM_BUS_PWRMGR_EVENT_MODECHANGED, &eventData , 0);
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("powerLedOn"), _T("{\"index\": \"power_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("powerLedOn"), _T("{\"index\": \"record_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("powerLedOn"), _T("{\"index\": \"data_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+
+    indicatorList.impl = nullptr;
+
+}
+
+TEST_F(FrontPanelDsTest, set24HourClock)
+{
+    device::FrontPanelTextDisplay displayList;
+    displayList.impl = &frontPanelTextDisplayImplStringMock;
+    displayList.FrontPanelIndicator::impl = &frontPanelIndicatorImplMock;
+
+    std::string test = "Text";
+
+    ON_CALL(frontPanelConfigImplMock, getTextDisplay(test))
+        .WillByDefault(::testing::ReturnRef(displayList));
+    ON_CALL(frontPanelTextDisplayImplStringMock, getCurrentTimeFormat())
+        .WillByDefault(::testing::Return(dsFPD_TIME_24_HOUR));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("set24HourClock"), _T("{\"is24Hour\": true}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+    displayList.impl = nullptr;
+    displayList.FrontPanelIndicator::impl = nullptr;
+
```

Comment:
I don't think such tests make sense

Reply:
please add expectation for setTimeFormat. getCurrentTimeFormat may not be called - it's used for logging which is not important

Reply:
@npoltorapavlo Pushed an updated for everything that was pointed out, please review when you have a chance.